source/code/plugins/agent_maintenance_script.rb

```diff
@@ -190,9 +191,44 @@ def apply_dsc_endpoint(server_resp)
       # When apply_dsc_endpoint is called from onboarding, dsc_endpoint will be returned in file
       update_config("DSC_ENDPOINT", dsc_endpoint)
 
+      generate_dsc_metaconfig
+
       return dsc_endpoint
     end
 
+    # Generate the DSC MetaConfig file. If the dsc endpoint is updated, DSC will
+    # not directly pick the endpoint from omsadmin.conf (instead, it is picked
+    # from MetaConfig.mof), so we must manually trigger metaconfig (re)generation.
+    def generate_dsc_metaconfig
```

Comment:
Umm will it more efficient to check if it has changed, rather than killing and restarting the process each time a topology response is received i.e. every hour?

Reply:
the process will not be killed

Reply:
Oh I see, you are running the command and monitoring that. my bad.

Reply:
No worries—basically we are running a .py script that will write to `generated_meta_config.mof`. Neither DSC nor OMS needs to be restarted for the change to that file to take effect. Since we run this as a subprocess, we need to make sure it doesn't  hang or turn zombie. For instance, if someone else is has already taken the DSC lock, this call could fail or take some time. In that case we shouldn't wait forever. Next time topology request is made we will try again.

Reply:
I think the mof file to be checked if the change is successfully applied is at `/etc/opt/omi/conf/omsconfig/configuration/MetaConfig.mof`

`generated_meta_config.mof` is an intermediate file generated which is sent to dsc while applying meta config.